llvm/lib/Target/AMDGPU/AMDGPURewriteUndefForPHI.cpp

```diff
@@ -177,6 +177,20 @@ bool AMDGPURewriteUndefForPHI::runOnFunction(Function &F) {
   return rewritePHIs(F, UA, DT);
 }
 
+PreservedAnalyses
+AMDGPURewriteUndefForPHIPass::run(Function &F, FunctionAnalysisManager &AM) {
+  UniformityInfo &UA = AM.getResult<UniformityInfoAnalysis>(F);
+  DominatorTree *DT = &AM.getResult<DominatorTreeAnalysis>(F);
+  bool Changed = rewritePHIs(F, UA, DT);
+  if (Changed) {
+    PreservedAnalyses PA;
+    PA.preserveSet<CFGAnalyses>();
+    return PA;
+  }
+
+  return PreservedAnalyses::all();
+}
+
 FunctionPass *llvm::createAMDGPURewriteUndefForPHIPass() {
```

Comment:
Should rename all the legacy pass bits to end with "LegacyPass"

Reply:
Should the class "AMDGPURewriteUndefForPHI" be renamed "AMDGPURewriteUndefForPHILegacy" or "AMDGPURewriteUndefForPHILegacyPass"? If the latter, we will end up with a function named "initializeAMDGPURewriteUndefForPHILegacyPassPass".

Reply:
AMDGPURewriteUndefForPHILegacy, some places have always been stuck with "PassPass"